### PR TITLE
doc: Added ABI compatibility doc page

### PR DESCRIPTION
**Pull request by FrancescoSer**
Added ABI compatibility doc page.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/abi_compatibility.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+* Function-calling conventions
+* Data structure layouts in memory
+* Exception handling mechanisms
+* Register usage conventions
+
+When ABI compatibility is maintained, binaries of one component can interface correctly with another without needing recompilation.
+For example, adding a new function to a library is typically an ABI-compatible change, as existing binaries remain functional.
+However, changes that affect data structure layouts, such as altering field order or size, break ABI compatibility as they change the memory layout expected by existing binaries.
+
+ABI Compatibility Matrix for the for the nrf54h20_soc_binary bundle
```
Comment (hubertmis) [inline doc/nrf/releases_and_maturity/abi_compatibility.rst:21]: I would add a paragraph to this section explaining that ABI compatibility for nrf54h20_soc_binary means that the System Controller or Secure Domain binaries do not need to be recompiled when the Application or Radio binary gets recompiled, as long as the Application and Radio are based on compatible NCS versions described in this section.

File: doc/nrf/releases_and_maturity/abi_compatibility.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+The table below illustrates ABI compatibility between different versions of the *nrf54h20_soc_binary* bundle and the |NCS|:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 25 25
+
+   * - nrf54h20_soc_binary version
+     - Compatible |NCS| versions
+     - Notes
```
Comment (hubertmis) [inline doc/nrf/releases_and_maturity/abi_compatibility.rst:32]: I think that from an application developer perspective, this table should be structured the other way around:

| NCS version | nrf54h20_soc_binary compatible with NCS version |

The entry point for an application developer is the NCS version they select for their application. The required nrf54h20_soc_binary is a result of their NCS version choice.

Comment (FrancescoSer) [reply doc/nrf/releases_and_maturity/abi_compatibility.rst:32]: Should we also mention the compatible nRF4H20 DK version?

Comment (hubertmis) [reply doc/nrf/releases_and_maturity/abi_compatibility.rst:32]: Hopefully not. When FP1.1 gets released all DKs should be equally compatible with the soc_bundle.

But someone from Aurora and Tarpand should confirm my hopes :)

**Review by hubertmis — APPROVED** [review]

File: doc/nrf/releases_and_maturity/abi_compatibility.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
+.. _abi_compatibility:
+
+ABI Compatibility
+*****************
+
+.. contents::
+   :local:
+   :depth: 2
+
+Application Binary Interface (ABI) compatibility defines how software components, like libraries and applications, interacts at the machine code level through their low-level binary interface, including:
```
Comment (annwoj) [inline doc/nrf/releases_and_maturity/abi_compatibility.rst:10]:
```suggestion
Application Binary Interface (ABI) compatibility defines how software components, like libraries and applications, interact at the machine code level through their low-level binary interface, including:
```

File: doc/nrf/releases_and_maturity/abi_compatibility.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+However, changes that affect data structure layouts, such as altering field order or size, break ABI compatibility as they change the memory layout expected by existing binaries.
+
+ABI Compatibility Matrix for the for the nrf54h20_soc_binary bundle
+===================================================================
+
+The table below illustrates ABI compatibility between different versions of the *nrf54h20_soc_binary* bundle and the |NCS|:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 25 25
```
Comment (annwoj) [inline doc/nrf/releases_and_maturity/abi_compatibility.rst:28]:
```suggestion
```

File: doc/nrf/releases_and_maturity/abi_compatibility.rst (hunk 2 above)
Comment (FrancescoSer) [reply doc/nrf/releases_and_maturity/abi_compatibility.rst]: @hakonfam FYI

**Review by greg-fer — COMMENTED** [review]
If this page is placed under `releases_and_maturity`, please make sure to check with the NCS doc release coordinators to include this page in the https://nordicsemi.atlassian.net/wiki/spaces/TECHDOC/pages/120300805/Tasks+required+for+a+release list.

File: doc/nrf/releases_and_maturity/abi_compatibility.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+* Register usage conventions
+
+When ABI compatibility is maintained, binaries of one component can interface correctly with another without needing recompilation.
+For example, adding a new function to a library is typically an ABI-compatible change, as existing binaries remain functional.
+However, changes that affect data structure layouts, such as altering field order or size, break ABI compatibility as they change the memory layout expected by existing binaries.
+
+ABI Compatibility Matrix for the for the nrf54h20_soc_binary bundle
+===================================================================
+
+The table below illustrates ABI compatibility between different versions of the *nrf54h20_soc_binary* bundle and the |NCS|:
```
Comment (annwoj) [inline doc/nrf/releases_and_maturity/abi_compatibility.rst:24]:
```suggestion
The following table illustrates ABI compatibility between different versions of the ``nrf54h20_soc_binary`` bundle and the |NCS|:
```

**Review by annwoj — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]

**Pull request merged**